celery/canvas.py

```diff
@@ -395,23 +395,37 @@ def __or__(self, other):
             other = maybe_unroll_group(other)
             if isinstance(self, _chain):
                 # chain | group() -> chain
-                sig = self.clone()
-                sig.tasks.append(other)
-                return sig
+                tasks = [t.clone() for t in self.tasks]
+                link_error = self.options.get('link_error', [])
```

Comment:
Glad you agreed with the suggestions :smile: ! One question here, is `link_error` a list of functions?

Reply:
Yes, `link_error` is list of signatures.
I've wrote it on your code suggestion ;)